CRM/Core/BAO/FinancialTrxn.php

```diff
@@ -50,37 +50,38 @@ public function __construct() {
    * @param array $params
    *   (reference ) an assoc array of name/value pairs.
    *
-   * @param string $trxnEntityTable
-   *   Entity_table.
+   * @param array $ids
+   *   The array that holds all the DB ids.
    *
    * @return CRM_Core_BAO_FinancialTrxn
    */
-  public static function create(&$params, $trxnEntityTable = NULL) {
+  public static function create(&$params, $ids = array()) {
     $trxn = new CRM_Financial_DAO_FinancialTrxn();
     $trxn->copyValues($params);
 
     if (!CRM_Utils_Rule::currencyCode($trxn->currency)) {
       $trxn->currency = CRM_Core_Config::singleton()->defaultCurrency;
     }
 
-    $trxn->save();
+    //per http://wiki.civicrm.org/confluence/display/CRM/Database+layer we are moving away from $ids array
+    $financialTrxnID = CRM_Utils_Array::value('FinancialTrxn', $ids, CRM_Utils_Array::value('id', $params));
```

Comment:
ie. just CRM_Utils_Array::value('id', $params)) is fine & we don't need to specially set it - copyValues will do that